src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -305,8 +305,8 @@ public bool Equals(RuntimeGraph? other)
                 return true;
             }
 
-            return Runtimes.OrderedEquals(other.Runtimes, pair => pair.Key, StringComparer.Ordinal)
-                && Supports.OrderedEquals(other.Supports, pair => pair.Key, StringComparer.Ordinal);
+            return Runtimes.ElementsEqual(other.Runtimes, pair => pair)
```

Comment:
These are Dict/Sets, should we an equals utility for dict/sets? This would avoid the extra dic allocation.

Reply:
I think that feedback applies to a few more types.

Reply:
Happy to add more overloads if we want the extra code. In addition to the cases you mentioned, There are a handful of calls using `IReadOnlyList<T>` that bind to the `IEnumerable<T>` overload which causes inefficient allocation of the intermediate collections.

Reply:
These comparisons happen often enough where it's worth it.

Reply:
Clarification question for you. The current implementation (here and in other places) compares keys using a specified string comparer. However, this could lead to unexpected issues if the dictionaries that are compared are constructed using different compareres. for example:

    var dictionaryA = new Dictionary<string, bool>(StringComparer.Ordinal);
    dictionaryA.Add("foo", true); // ContainsKey("FOO") returns false
    var dictionaryB = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
    dictionaryB.Add("FOO", true); // ContainsKey("foo") returns true

If we use the existing implementation of `OrderedEquals`, the dictionaries are considered different. If we Enumerate the keys of one dictionary and try to do a lookup in the other dictionary, the order we do that in matters. Curious about what you think the expected behavior is for NuGet's scenarios.

Reply:
I'd expect we're not trying to compare dictionaries/hashsets with different comparers. Those would be bugs. 

Should we compare the comparers in the equals method?

Reply:
Sounds good. I was hoping to verify that those cases are unexpected.

As far as checking comparers, the exposed type is `IReadOnlyDictionary<TKey, TValue>`. I can check the underlying type to see if it has the comparer exposed. Looks like the underlying type is `Dictionary<TKey, TValue>`, so that should work.